scripts/bulk_update_leads_info.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import re
+import argparse
+import csv
+import logging
+from closeio_api import Client as CloseIO_API, APIError
+
+OPPORTUNITY_FIELDS = ['opportunity%s_note',
+                      'opportunity%s_value',
+                      'opportunity%s_value_period',
+                      'opportunity%s_confidence',
+                      'opportunity%s_status']
+
+
+def get_contact_info(contact_no, csv_row, what, contact_type):
+    columns = [x for x in csv_row.keys()
+               if re.match(r'contact%s_%s[0-9]' % (contact_no, what), x) and csv_row[x]]
+    contact_info = []
+    for col in columns:
+        contact_info.append({what: csv_row[col], 'type': contact_type})
+    return contact_info
+
+parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter, description="""
+Imports leads and related data from a csv file with header.
+Header's columns may be declared in any order. Detects csv dialect (delimeter and quotechar).
+""", epilog="""
+key columns:
+    * lead_id                           - If exists and not empty, update using lead_id.
+    * company                           - If lead_id is empty or is not exists, imports to
+                                          first lead from found company. If the company was
+                                          not found, loads as new lead.
+lead columns:
+    * url                               - lead url, may be empty
+    * description                       - lead description, may be empty
+    * note[0-9]                         - lead notes, may be empty
```

Comment:
since all fields "may be empty" i don't think you should write this on each line

also please note here that this *creates* new notes (versus updating existing)